contracts/apps/accounting/AccountingApp.sol

```diff
@@ -0,0 +1,133 @@
+pragma solidity ^0.4.11;
+
+import "../Application.sol";
+
+contract AccountingApp is Application {
+
+    // The concept of sending tokens to or from the org
+    struct Transaction {
+        address token;
+        int value;
+        address baseToken;
+        int baseValue;
+        address externalAddress;
+        string reference;
+        uint timestamp;
+    }    
+
+    // The state a transaction update can be.
+    // New states should be added to the end to maintain the 
+    // order of the index when interfacing with web3.
+    enum TransactionState {
+        New, // not needed?
+        PendingApproval,
+        Failed,
+        Succeeded
+    }
+
+    // The change in Transaciton state over time
+    struct TransactionUpdate {
+        uint transactionId; // Parent Transaction
+        TransactionState state;
+        string reason;
+        address actor; // who performed this update 
+    }    
+
+    Transaction[] public transactions;
+    TransactionUpdate[] public transactionUpdates;
+
+    // Reverse relation of a Transaction ID  -> TransactionsUpdatesIds[]
+    // transactionUpdatesRelation[tid] = [tuid_0..tuid_N]
+    mapping (uint => uint[]) public transactionUpdatesRelation;
+
+    // Create a new transaction and return the id of the new transaction.
+    // externalAddress is where the transication is coming or going to.
+    bytes4 constant NEW_TRANSACTION_SID = bytes4(sha3('newTransaction(address token, int value, address externalAddress, string reference, TransactionState initialState)'));
```

Comment:
I like this right above the function.

Minor detail function signatures are calculated without the parameter names `sha3('newTransaction(address,int..`